packages/base/src/3dview/mainview.tsx

```diff
@@ -1455,6 +1457,20 @@ export class MainView extends React.Component<IProps, IStates> {
         );
       }
     }
+
+    if (change.key === 'transform') {
+      const transformEnabled = change.newValue as boolean | undefined;
+
+      if (transformEnabled !== undefined) {
+        this.setState(
+          old => ({ ...old, transform: transformEnabled }),
+          () => {
+            this._updateSelected({});
```

Comment:
Is this right? Shouldn't we just set the transformControls object's `visible` attribute here instead? Or the `enabled` property

Reply:
yes, i tried that but just setting those was preventing transformControls from disabling or enabling right on clicking the toggle button, which is why I also had to trigger this part of code: 

https://github.com/jupytercad/JupyterCAD/blob/3340d1d62c30b4c863dac74563bbaec19f26db5e/packages/base/src/3dview/mainview.tsx#L1181-L1196

Reply:
Hmm, though calling `_updateSelected({})` means unselecting anything that was previously selected.

I'm a bit confused you need this code to be triggered again. This code should already have been triggered upon the latest selection.

Reply:
> This code should already have been triggered upon the latest selection.

i am needed to reselect the object manually after toggling the transform controls from the button 

> though calling _updateSelected({}) means unselecting anything that was previously selected.

yes this is not ideal and should be done better, happy to hear thoughts

Reply:
> > i am needed to reselect the object manually
> 
> Do you know why you need that?

It was most probably needed because of missing attach & detach calls.
Is it better after [`f0b2c6d`](https://github.com/jupytercad/JupyterCAD/pull/578/commits/f0b2c6d28af1a0ee2bd271f68c37718a7c26b8a5)? if not what should be the right method to get the selected object?

Reply:
That does not look correct to me `this._pick()` will run the raycast picking, so it's entirely dependent on your current mouse position. This means your code does not work (won't attach to the correct object) if the mouse is mal-positioned.

I'm still not sure why you need to attach/detach here?

[It should already have been attached here on the latest selection](https://github.com/jupytercad/JupyterCAD/blob/9b4160bf274697db9783892dd56010b315399f0a/packages/base/src/3dview/mainview.tsx#L1185). So AFAIU your code here should only do `enabled = false/true` and `visible = false/true`.

Reply:
Ahh, now it works without attach/detach calls, i must not have been doing it 100% correctly before

Reply:
Thanks for checking this out!